src/robotFactory.js

```diff
@@ -1,17 +1,67 @@
 'use strict';
 
-function BaseRobot() {
-  // implement
+function BaseRobot(name, weight, coords, chipVersion) {
+  this.name = name;
+  this.weight = weight;
+  this.coords = coords;
+  this.chipVersion = chipVersion;
 }
 
-function FlyingRobot() {
-  // implement
+BaseRobot.prototype = {
+  goForward(step = 1) {
+    this.coords.y += step;
+  },
+  goBack(step = 1) {
+    this.coords.y -= step;
+  },
+  goRight(step = 1) {
+    this.coords.x += step;
+  },
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  },
+  getInfo() {
+    // eslint-disable-next-line max-len
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
```

Comment:
```suggestion
    return `Robot: ${this.name}, `
    + `Chip version: ${this.chipVersion}, `
    + `Weight: ${this.weight}`;
```
Let's not disable es-lint